### PR TITLE
Feature/add rocksdb build

### LINKED ISSUE #279: Problem: cronos binary don't support rocksdb backend
Built with rocksdb backend, so we can experiment how it perform.

### PR DESCRIPTION
**Pull request by allthatjazzleo**
fix #279

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,6 +20,16 @@ ldflags = -X github.com/cosmos/cosmos-sdk/version.Name=cronos \
 	-X github.com/cosmos/cosmos-sdk/version.AppName=cronosd \
 	-X github.com/cosmos/cosmos-sdk/version.Version=$(VERSION) \
 	-X github.com/cosmos/cosmos-sdk/version.Commit=$(COMMIT)
```
Comment (yihuang) [inline Makefile:22]: add this too?
```
-X "github.com/cosmos/cosmos-sdk/version.BuildTags=$(build_tags_comma_sep)" \
```

So we can see the supported db backends in help screen.

Comment (allthatjazzleo) [reply Makefile]: fixed

File: Makefile [hunk 2]
```diff
@@ -47,18 +41,59 @@ ifeq ($(LEDGER_ENABLED),true)
...
+ifeq (boltdb,$(findstring boltdb,$(COSMOS_BUILD_OPTIONS)))
+  build_tags += boltdb
+  ldflags += -X github.com/cosmos/cosmos-sdk/types.DBBackend=boltdb
+endif
+
+build_tags += $(BUILD_TAGS)
+build_tags := $(strip $(build_tags))
+
+whitespace :=
+whitespace += $(whitespace)
```
Comment (yihuang) [inline Makefile:78]:
```suggestion
whitespace := $(subst ,, )
```

**Review by yihuang — APPROVED** [review]
The `build_tags_comma_sep`'s comma seems not correct, otherwise LGTM.

File: go.mod [hunk 3]
```diff
@@ -147,18 +147,21 @@ require (
...
-replace github.com/99designs/keyring => github.com/crypto-org-chain/keyring v1.1.6-fixes
+	// TODO: remove when middleware will be implemented
+	github.com/cosmos/ibc-go/v2 => github.com/crypto-org-chain/ibc-go/v2 v2.0.1-hooks
 
-// TODO: remove when middleware will be implemented
-replace github.com/cosmos/ibc-go/v2 => github.com/crypto-org-chain/ibc-go/v2 v2.0.1-hooks
+	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
 
-replace github.com/cosmos/iavl => github.com/cosmos/iavl v0.17.3
+	github.com/tecbot/gorocksdb => github.com/cosmos/gorocksdb v0.0.0-20211202124722-2c356d6d98e4
```
Comment (thomas-nguy) [inline go.mod:161]: can you add a comment on why it is needed?

**Review by thomas-nguy — APPROVED** [review]

File: gomod2nix.toml [hunk 4]
```diff
@@ -3602,12 +3602,13 @@
     sha256 = "11b4vjldnp6ws52wlqrzsy6nnn5qa2m98wiqvd90nzsh0n3bi9nz"
 
 ["github.com/tecbot/gorocksdb"]
```
Comment (JayT106) [inline gomod2nix.toml:3604]: should be `github.com/cosmos/gorocksdb`

File: gomod2nix.toml [hunk 5]
```diff
@@ -3602,12 +3602,13 @@
...
-  sumVersion = "v0.0.0-20191217155057-f0fad39f321c"
+  sumVersion = "v0.0.0-20211202124722-2c356d6d98e4"
+  vendorPath = "github.com/cosmos/gorocksdb"
   ["github.com/tecbot/gorocksdb".fetch]
```
Comment (JayT106) [inline gomod2nix.toml:3607]: same here.

File: gomod2nix.toml (hunk 4 above)
Comment (yihuang) [reply gomod2nix.toml:3604]: this file is generated automatically by tool `gomod2nix`, so it's fine.

**Pull request merged**